cost-analyzer/values.yaml

```diff
@@ -536,6 +536,40 @@ kubecostFrontend:
 #    fqdn: kubecost-multi-diag.kubecost.svc.cluster.local:9007
 #  clusterController:
 #    fqdn: cluster-controller.kubecost.svc.cluster.local:9731
+#
+
+  # Configurable headers for nginx responses.
+  nginxHeaders:
+    # applied to all route locations
+    server:
+      - Content-Security-Policy "default-src 'self' cdn.userway.org; frame-ancestors 'none';";
+      - Cache-Control "must-revalidate";
+    # per-location headers
+    location:
+      root: []
+      healthz: []
+      model: []
+      clusterTurndown:
+        - "'Access-Control-Allow-Origin' '*' always;"
+        - "'Access-Control-Allow-Methods' 'GET, PUT, POST, DELETE, OPTIONS' always;"
+      hideOrphanedResources:
+        - "'Access-Control-Allow-Origin' '*' always;"
+        - "'Access-Control-Allow-Methods' 'GET, PUT, POST, DELETE, OPTIONS' always;"
+      hideDiagnostics:
+        - "'Access-Control-Allow-Origin' '*' always;"
+        - "'Access-Control-Allow-Methods' 'GET, PUT, POST, DELETE, OPTIONS' always;"
+      multiClusterDiagnosticsEnabled:
+        - "'Access-Control-Allow-Origin' '*' always;"
+        - "'Access-Control-Allow-Methods' 'GET, PUT, POST, DELETE, OPTIONS' always;"
+      multiClusterDiagnostics:
+        - "'Access-Control-Allow-Origin' '*' always;"
+        - "'Access-Control-Allow-Methods' 'GET, PUT, POST, DELETE, OPTIONS' always;"
+      forecastingEnabled:
+        - "'Access-Control-Allow-Origin' '*' always;"
+        - "'Access-Control-Allow-Methods' 'GET, PUT, POST, DELETE, OPTIONS' always;"
+      productConfigs:
+        - "'Access-Control-Allow-Origin' '*' always;"
+        - "'Access-Control-Allow-Methods' 'GET, PUT, POST, DELETE, OPTIONS' always;"
```

Comment:
I'm worried that one configuration per route could quickly become unwieldy. Would it be possible start with:

```yaml
nginxHeaders:
  routes:
    - "'Access-Control-Allow-Origin' '*' always;"
    - "'Access-Control-Allow-Methods' 'GET, PUT, POST, DELETE, OPTIONS' always;"
```

These headers would then be added to all routes. Do you see an issue with this from a frontend or security perspective?

Reply:
I don't think that's an issue. I was attempting to keep the same behavior we have today, where some `location` directives don't support CORS...personally I have no problem with broadly applying it! I can update.